File: levelOrder.java
import java.util.Queue;

class binaryTree{
	private int data;
	private binaryTree left;
	private binaryTree right;

	public binaryTree insert(binaryTree root, int data){
	
		if(root == null){
			binaryTree temp = new binaryTree();
                	temp.left = null;
                	temp.right = null;
                	temp.data = data;
			root = temp;
		}else{
			if(root.data > data)
				root.left = insert(root.left, data);
			if(root.data < data)	
				root.right = insert(root.right, data);
		}
		return root;
	}
	public void levelOrder(binaryTree root){
		Queue<binaryTree> queue = new LinkedList<binaryTree>();
		queue.add(root);
		queue.add(null);
		while(queue.)
		 
	}
	public void display(binaryTree root){
		binaryTree temp = new binaryTree();
		temp = root;
		if(root == null) return ;

		if(root.left != null) display(root.left);

		System.out.println("data->" + root.data);

		if(root.right != null) display(root.right);
	}
	public static void main(String args[]){
		binaryTree root = new binaryTree();
		binaryTree obj = new binaryTree();
		binaryTree par = new binaryTree();
		root = null;
		par  = null; 	
		root = obj.insert(root, 5);
		root = obj.insert(root, 6);
		root = obj.insert(root, 1);
		root = obj.insert(root, 4);
		root = obj.insert(root, 8);
                root = obj.insert(root, 9);
		root = obj.insert(root, 10);
                root = obj.insert(root, 14);
		obj.display(root);

		
	}
} 

File: levelOrder.c
#include<stdio.h>
#include<stdlib.h>

struct tree{
	struct tree *right;
	struct tree *left;
	int data;

}*root;


void  insert(struct tree **root , int data){

	if(*root == NULL){
		struct tree *temp = malloc(sizeof(struct tree));
		temp->right = NULL;
		temp->left  = NULL;
		temp->data  = data;
		*root = temp;
		return ;	
	}else{
		if((*root)->data > data){
			 insert(&(*root)->left,data);
		}else if ((*root) ->data < data){
			 insert(&(*root)->right, data);
		}	

	     }
}

void levelorder(struct tree *root, struct tree *level){                                                                                                        
	 int i=-1,j=1;
	                                                                                                                                               if(root ==  NULL){
		return ;
	}else{
		while(level[i] == level[j]){
			if(level[i+1] = = 1000){
				level[j++] = 1000;
				root = level[i++];
			
			}else{
				root = level[i++];
			}
			if(root->left != NULL){
				level[j++] = root->left;
			}
			if(root->right != NULL){	
                        	level[j++] = root->right;
			}	
			
		}
	}
}		

void  displayLVR(struct tree *root){
	if(root == NULL){
		return ;
	}else{
		displayLVR(root->left);
		printf("%d\t",root->data);
		displayLVR(root->right);
	}
}

int main(){
	root = NULL;
	insert(&root,5);
	insert(&root,6);
	insert(&root,1);
	insert(&root,4);
	insert(&root,9);
	insert(&root,10);
	insert(&root,14);
	insert(&root,8);
	displayLVR(root);
	struct tree *level;
	level[0] = root;
	level[1] = "flag";
	levelorder(root,level);
	printf("\nlevelorder of tree\n");
	displayLVR(root);
	return 0;
}
